client/src/js/samples/components/Create/Create.js

```diff
@@ -32,206 +33,203 @@ const CreateSampleFields = styled.div`
     grid-column-gap: ${props => props.theme.gap.column};
 `;
 
+const StyledInputError = styled(InputError)`
+    text-align: left;
+`;
+
 const extensionRegex = /^[a-z0-9]+-(.*)\.f[aq](st)?[aq]?(\.gz)?$/;
 
-const getFileNameFromId = id => id.match(extensionRegex)[1];
-
-const getInitialState = props => ({
-    selected: [],
-    name: "",
-    host: "",
-    isolate: "",
-    locale: "",
-    subtractionId: "",
-    group: props.forceGroupChoice ? "none" : "",
-    errorName: "",
-    errorSubtraction: "",
-    errorFile: "",
-    libraryType: "normal"
+// This method's inputs are the selected read file's id and the list of read files
+// The output is the id's corresponding name without the file extension
+const getFileNameFromId = (id, files) => {
+    const file = find(files, file => file.id === id);
+    return file ? file.name_on_disk.match(extensionRegex)[1] : "";
+};
+
+const validationSchema = Yup.object().shape({
+    name: Yup.string().required("Required Field"),
+    subtractionId: Yup.string().required("A default subtraction must be selected"),
+    selected: Yup.array().min(1, "At least one read file must be attached to the sample")
 });
 
-export class CreateSample extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = getInitialState(props);
-    }
-
-    static getDerivedStateFromProps(nextProps, prevState) {
-        if (!prevState.errorName.length && !!nextProps.error) {
-            return { errorName: nextProps.error };
-        }
-
-        return null;
-    }
-
-    componentDidMount = () => {
-        this.props.onLoadSubtractionsAndFiles();
-    };
-
-    handleChange = e => {
-        const { name, value, error } = getTargetChange(e.target);
-
-        if (name === "name" || name === "subtractionId") {
-            this.setState({
-                [name]: value,
-                [error]: ""
-            });
-        } else {
-            this.setState({
-                [name]: value
-            });
-        }
+export const CreateSample = props => {
+    const [reinitialize, setReinitialize] = useState(true);
+
+    const initialValues = {
+        selected: [],
+        name: "",
+        isolate: "",
+        host: "",
+        locale: "",
+        libraryType: "normal",
+        selected: [],
+        // Values below will be updated on mount since they are dependent on props
+        group: "",
+        subtractionId: ""
     };
 
-    handleLibrarySelect = libraryType => {
-        this.setState({ libraryType });
+    const updateInitialValues = () => {
+        initialValues.group = props.forceGroupChoice ? "none" : "";
+        initialValues.subtractionId = get(props, "subtractions[0].id", "");
+        // Prevent further reinitialization of the form
+        setReinitialize(false);
     };
```

Comment:
This is a major no-no.

In React, all data other than constants **must** live in `props` or `state`. You have data in `initialValues` that is then being mutated by a callback. Never do this.

Please resolve the other comments in the review. We should discuss this one on a call.